sonar-text-plugin/src/main/java/org/sonar/plugins/secrets/api/PatternMatcher.java

```diff
@@ -46,16 +47,26 @@ public static PatternMatcher build(@Nullable Matching matching) {
     return new PatternMatcher(null);
   }
 
+  public static PatternMatcher build(AuxiliaryPattern auxiliaryPattern) {
+    return new PatternMatcher(auxiliaryPattern.getPattern());
+  }
+
   public List<Match> findIn(String content) {
     if (pattern == null) {
       return Collections.emptyList();
     }
     List<Match> matches = new ArrayList<>();
     Matcher matcher = pattern.matcher(content);
+
     while (matcher.find()) {
       MatchResult matchResult = matcher.toMatchResult();
-      matches.add(new Match(matchResult.group(1), matchResult.start(1), matchResult.end(1)));
+      if (matcher.groupCount() == 0) {
+        matches.add(new Match(matchResult.group(), matchResult.start(), matchResult.end()));
+      } else {
+        matches.add(new Match(matchResult.group(1), matchResult.start(1), matchResult.end(1)));
+      }
```

Comment:
I hope this doesn't break anything. WDYT?

Reply:
The only thing I can see for now is that we'll lose a kind of validation that we were implying that main matches will contain at least one capture group. Shouldn't _break_ anythigh though, as far as I can see.